src/main/scala/com/snowplowanalytics/righttobeforgotten/RightToBeForgottenSparkJob.scala

```diff
@@ -0,0 +1,75 @@
+/**
+  * Copyright 2018 Snowplow Analytics
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *     http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package com.snowplowanalytics.righttobeforgotten
+
+import org.apache.spark.sql.{Column, DataFrame, Dataset, SparkSession}
+import org.apache.spark.sql.functions.{col, not, split}
+
+class RightToBeForgottenSparkJob(r2fConf: ValidatedR2FJobConfiguration) {
+  val inputDirectory = r2fConf.inputDirectory
+  val outputMatchingDirectory = r2fConf.outputMatchingDataDirectory
+  val outputNonMatchingDirectory = r2fConf.outputNonMatchingDataDirectory
+
+  def run(spark: SparkSession): Unit = {
+    val input = spark.read.textFile(inputDirectory)
+    val filterColumnDf: DataFrame = getFilterColumnDF(input)
+    val filterColumn = col(RightToBeForgottenSparkJob.FilterMatchedColumn)
+    val matchingRows =
+      getFilteredDataframe(filterColumnDf, filterColumn)
+    val nonMatchingRows =
+      getFilteredDataframe(filterColumnDf, not(filterColumn))
+    val matchingRowsCount = matchingRows.rdd.count().toDouble
+    val nonMatchingRowsCount = nonMatchingRows.rdd.count().toDouble
```

Comment:
I think I remember noticing this, but I don't remember what was the reason not to do `df.count()`?

Reply:
Its required due to a bug in spark as far as I can tell.

Reply:
As far as I can tell this fails due to a spark bug and dropping to rdd is required.